keyrings/alt/pyfs.py

```diff
@@ -47,24 +75,72 @@ class BasicKeyring(KeyringBackend):
     def __init__(self, crypter, filename=None, can_create=True, cache_timeout=None):
         super(BasicKeyring, self).__init__()
         self._crypter = crypter
-        def_fn = os.path.join(platform_.data_root(), self.__class__._filename)
-        self._filename = filename or def_fn
+        if filename is None:
+            self._filename = os.path.join(
+                platform_.data_root(), self.__class__._filename
+            )
+        elif filename.endswith("/"):
```

Comment:
This code seems to be changing the meaning of the filename parameter from being a filename to being a filename_or_directory, but only treating it like a directory if it ends in a posix-like slash. I don't care for that interface and I'm not sure what problem this solves. Probably this change should have its own issue and separate change.